### ISSUE TITLE
XMLHttpRequest/send-entity-body-document.htm intermittently fails

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#8157]

**Comment by jdm** [comment]
```
plain text file
---------------
Expected PASS, got FAIL
assert_equals: document should be serialized and sent as text/html;charset=UTF-8 (TEST#3) expected "text/html;charset=UTF-8" but got "text/plain;charset=UTF-8"
request/<@http://web-platform.test:8000/XMLHttpRequest/send-entity-body-document.htm:18:1
Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1379:20
test@http://web-platform.test:8000/resources/testharness.js:495:9
request@http://web-platform.test:8000/XMLHttpRequest/send-entity-body-document.htm:13:1
init@http://web-platform.test:8000/XMLHttpRequest/send-entity-body-document.htm:29:42
load@http://web-platform.test:8000/XMLHttpRequest/send-entity-body-document.htm:0:1
```

This line from a bit earlier in the log is plausibly related:
```
 1:20.60 LOG: Thread-1511 wptserve WARNING Traceback (most recent call last):
  File "/Users/servo/buildbot/slave/mac-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptserve/wptserve/handlers.py", line 244, in __call__
    rv = self.func(request, response)
  File "/Users/servo/buildbot/slave/mac-rel-wpt/build/tests/wpt/web-platform-tests/XMLHttpRequest/resources/form.py", line 2, in main
    return "id:%s;value:%s;" % (request.POST.first("id"), request.POST.first("value"))
  File "/Users/servo/buildbot/slave/mac-rel-wpt/build/tests/wpt/web-platform-tests/tools/wptserve/wptserve/request.py", line 505, in first
    raise KeyError
KeyError
```

**Comment by Ms2ger** [comment]
It's not related; `form.py` is only used in `send-data-formdata.htm`.

**Comment by mbrubeck** [comment]
Latest failure was on Linux.

**Comment by eefriedman** [comment]
This should be fixed by #6677.  (By "fixed" here, I mean the test will fail consistently.  The test is badly structured, so the testrunner skips some of the tests when the load event fires early.)

**Comment by jdm** [comment]
Fixed by #6677.
